File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    let maxSeq = 0
    let isExist = {}
    
    for (let i = 0; i < nums.length; i++) {        
        let seq = 1;
        isExist[nums[i]] = true
        
        let itr = nums[i]
        while ( isExist[++itr] != undefined ) seq++;
        itr = nums[i]
        while ( isExist[--itr] != undefined ) seq++;
        
        
        if (seq > maxSeq) maxSeq = seq;
    }
    return maxSeq;
};

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        set <int> hashset;
        
        for(int num: nums) {
            hashset.insert(num);
        }
        
        int maxSeq = 0;
        
        for (int num: nums) {
            
            if ( !hashset.count(num - 1)) {
                int seq = 1;

                while ( hashset.count(num + 1) ) {
                    num += 1;
                    seq += 1;
                }
                
                maxSeq = max(seq, maxSeq);
            }
        }
        return maxSeq;
    }
};